crossversion/xversion.py

```diff
@@ -454,6 +454,21 @@ def run_test(bld_server, bld_client, test_client=False, test_tool=False, test_ch
             invalid_tool_pairs.append([bld.branch,"v3.0"])
             invalid_tool_pairs.append([bld.branch,"v2.2"])
             invalid_tool_pairs.append([bld.branch,"v2.0"])
+        elif target_branch == "master":
+            invalid_pairs.append([bld.branch,"master"])
+            invalid_tool_pairs.append([bld.branch,"master"])
+            invalid_pairs.append(["master", bld.branch])
+            invalid_tool_pairs.append(["master", bld.branch])
+        elif target_branch == "v4.2":
+            invalid_pairs.append([bld.branch,"v4.2"])
+            invalid_tool_pairs.append([bld.branch,"v4.2"])
+            invalid_pairs.append(["v4.2", bld.branch])
+            invalid_tool_pairs.append(["v4.2", bld.branch])
+        elif target_branch == "v4.1:
+            invalid_pairs.append([bld.branch,"v4.1"])
+            invalid_tool_pairs.append([bld.branch,"v4.1"])
+            invalid_pairs.append(["v4.1", bld.branch])
+            invalid_tool_pairs.append(["v4.1", bld.branch])
```

Comment:
Why not make a more general block outside of the selection logic like:
```python
          # Do not test the Build branch against the Target branch.
          # If we are fixing cross-version issues then this is likely to fail which is correct in this circumstance.
          invalid_pairs.append([bld.branch, target_branch])
          invalid_pairs.append([target_branch, bld.branch])
          invalid_tool_pairs.append([bld.branch, target_branch])
          invalid_tool_pairs.append([target_branch, bld.branch])
```

Reply:
Yeah, I'm not entirely wild about this patch just yet. We don't want to test against the HEAD of the specific target branch as that is what we are changing, but we do want to test against any prior tagged releases in that branch to ensure we aren't breaking compatibility within the series. Still scratching my head over it. Your suggestion would indeed clean things up - just not sure yet what the logic should be to get what we really want.

Hope that makes some sense.